src/Eshopworld.WorkerProcess/CosmosDistributedLock/DistributedLock.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Threading.Tasks;
+
+namespace EShopworld.WorkerProcess.CosmosDistributedLock
+{
+    public class DistributedLock : IDistributedLock
+    {
+        private readonly ICosmosDistributedLockStore _cosmosDistributedLockStore;
+        private CosmosDistributedLockClaim _cosmosDistributedLockClaim;
+
+        public DistributedLock(ICosmosDistributedLockStore cosmosDistributedLockStore)
+        {
+            _cosmosDistributedLockStore = cosmosDistributedLockStore;
+        }
+
+        public async Task<IDisposable> Acquire(string lockName)
```

Comment:
You could add optional param here (and also have it in the config by default) of a wait timeout + cancelation token to break the wait

Reply:
(separate PR)